src/main/java/com/direwolf20/buildinggadgets/common/schema/template/TemplateData.java

```diff
@@ -0,0 +1,56 @@
+package com.direwolf20.buildinggadgets.common.schema.template;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.math.BlockPos;
+
+import java.util.Objects;
+
+/**
+ * This class is a mutable wrapper around {@link BlockPos position}, {@link BlockState state} and {@link CompoundNBT nbt}
+ * and therefore represents a single data point in a {@link Template}. Notice that because it is mutable this class should
+ * not be stored directly when outside code has references to it.
+ */
+public final class TemplateData {
+    private BlockPos pos;
+    private BlockState state;
+
+    public TemplateData(BlockPos pos, BlockState state) {
+        setInformation(pos, state);
+    }
+
+    public BlockPos getPos() {
+        return pos;
+    }
+
+    public BlockState getState() {
+        return state;
+    }
+
+    public TemplateData setInformation(BlockPos pos, BlockState state) {
```

Comment:
```suggestion
    public TemplateData of(BlockPos pos, BlockState state) {
```

or 
```suggestion
    public TemplateData setData(BlockPos pos, BlockState state) {
```

Reply:
Going to go for setData, as of indicates the creation of a new instance, which it doesn't do.